src/test/java/com/epam/university/java/core/task057/Task057Test.java

```diff
@@ -0,0 +1,68 @@
+package com.epam.university.java.core.task057;
+
+import com.epam.university.java.core.helper.TestHelper;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class Task057Test {
+    Task057 instance;
+
+    /**
+     * Instantiate class.
+     *
+     * @throws Exception in case of Runtime Exceptions.
+     */
+    @Before
+    public void setUp() throws Exception {
+        instance = TestHelper.getInstance(Task057.class);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void substringIsNull() {
+        String path = getClass().getResource("/task057/").getPath();
+        instance.find(path, null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void pathIsNull() {
+        instance.find(null, "test1");
+    }
+
+    @Test
+    public void oneFile() {
+        String path = getClass().getResource("/task057/").getPath();
+        List<String> result = instance.find(path, "test2");
+        assertTrue("Incorrect list size or result isn't contain target",
+                result.size() == 1
+                        && result.contains(getClass().getResource("/task057/test.txt").getPath()));
+    }
+
+    @Test
+    public void severalFiles() {
+        String path = getClass().getResource("/task057/").getPath();
+        String target1 = getClass().getResource("/task057/subdir1/test1.txt").getPath();
```

Comment:
I suppose, it's better using something like `Path.of()` and then `Path.resolve()` to join path's parts.

Reply:
I think fixed:) Please check